testsuite/tests/lib-dynlink-csharp/Makefile

```diff
@@ -30,7 +30,9 @@ all: prepare bytecode bytecode-dll native native-dll
 prepare:
 	@if $(SUPPORTS_SHARED_LIBRARIES); then \
```

Comment:
I'd suggest that the CSC be fixed in a separate commit independently from this PR since it's a problem with the existing code as well. This PR and commit is only about fixing the native/arch issue, and fixes failing builds. Fixing the CSC check is an optimisation, but it doesn't un-break the currently-failing architectures.